src/mono/mono/metadata/metadata.c

```diff
@@ -7805,7 +7805,7 @@ guint
 mono_aligned_addr_hash (gconstpointer ptr)
 {
 	/* Same hashing we use for objects */
-	return (GPOINTER_TO_UINT (ptr) >> 3) * 2654435761u;
```

Comment:
overflowing here is expected

Reply:
yeah I suspect we'd need to audit every usage if/when we turn on validation, but right now this gives us all the places where conversions are happening.